receiver/gitlabreceiver/config.go

```diff
@@ -0,0 +1,80 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package gitlabreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/gitlabreceiver"
+
+import (
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.uber.org/multierr"
+)
+
+const (
+	defaultReadTimeout  = 500 * time.Millisecond
+	defaultWriteTimeout = 500 * time.Millisecond
+	defaultPath         = "/events"
+	defaultHealthPath   = "/health"
+	defaultEndpoint     = "localhost:8080"
+)
+
+var (
+	errReadTimeoutExceedsMaxValue  = errors.New("the duration specified for read_timeout exceeds the maximum allowed value of 10s")
+	errWriteTimeoutExceedsMaxValue = errors.New("the duration specified for write_timeout exceeds the maximum allowed value of 10s")
+	errRequiredHeader              = errors.New("both key and value are required to assign a required_header")
+	errConfigNotValid              = errors.New("configuration is not valid for the gitlab receiver")
+)
+
+// Config that is exposed to this gitlab receiver through the OTEL config.yaml
+type Config struct {
+	WebHook WebHook `mapstructure:"webhook"`
+}
+
+type WebHook struct {
+	confighttp.ServerConfig `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+	Path                    string                   `mapstructure:"path"`            // path for data collection. Default is /events
+	HealthPath              string                   `mapstructure:"health_path"`     // path for health check api. Default is /health_check
+	RequiredHeader          RequiredHeader           `mapstructure:"required_header"` // optional setting to set a required header for all requests to have
+	Secret                  string                   `mapstructure:"secret"`          // secret for webhook
+}
+
+type RequiredHeader struct {
```

Comment:
I've been thinking about this for the GitHub receiver as well, but I think  this should be converted to `RequiredHeaders` of `map[string]string` so that a user can set any set of headers that are required. The main one that would be useful for [GitLab](https://docs.gitlab.com/ee/user/project/integrations/webhooks.html#delivery-headers) would be `X-Gitlab-Instance`. But if a user has a WAF sitting in front of the receiver (which they should anyway for public traffic) then they can set additional headers of their own choosing.

For the other headers listed in the doc, I think we should map these internal to the receiver.